views/pages/index.ejs

```diff
@@ -1,27 +1,31 @@
 <!DOCTYPE html>
 <html>
-    <head>
-        <% include ../partials/header.ejs %>
-        <!-- <script src="../../public/scripts/client.js"></script> -->
-        <script src="/client.js"></script>
-    </head>
-    <body>
-        <div class="wrapper">
-            <p class="header-text">What's Cooking, Good Looking?</p>
-            
-            <label for="id">ID</label>
-            <input type="text" id="id"/>
-            <label for="name">Name</label>
-            <input type="text" id="name"/> 
-            <!-- <label for="insert">Insert</label>
-            <input type="radio" id="insert" name="queryType" value="insert">
-            <label for="delete">Delete</label>
-            <input type="radio" id="delete" name="queryType" value="delete"> -->
-            <br>
-            <button onclick="addRow()">Insert Row</button>
-            <button onclick="deleteRow()">Delete Row</button>
-            <!-- <button onclick="updateRow()">Update Row</button> -->
-          
-        </div>
-    </body>
-</html>
+<head>
+<meta charset="UTF-8">
+<link rel = "stylesheet" href= "/stylesheets/search.css">
+<title>Main Page</title>
+</head>
+<body>
+ <div class="search-box">
+   <input type="text" name="" class="search-txt" placeholder="Search..."/>
+   <a class="search-btn" href = "#">
+   </a>
+  </div>
+<font color = #ffffff >
+<h1> Hello User!</h1>
+<p>Here's what is in your fridge </p>
+<table  border = 2 style="width:50%">
+  <tr>
+    <th>Products</th>
+    <th>Type</th>
+    <th>Date of Expiry</th>
+    <th>Quantity</th>
+  </tr>
+            <td><%= r.prod_name%></td>
+            <td><%= r.prod_type%></td>
+            <td><%= r.prod_expiry%></td>
+            <td><%= r.prod_quantity%></td>
```

Comment:
This is great, but it looks like there might be just a couple missing elements here. Firstly, the "results" variable passed to this page by the server needs to be accessed in a forEach loop first before you can access its fields using r.xxx in the table. For reference you can take a look at the db.ejs file which has a very similar operation. Adding the line `<% results.forEach(function(r) { %>` before the <td> elements will allow you to access the fields of results through the variable r, as you are doing here. After that don't forget to close the forEach loop using the closing tag `<% }); %>` below the <td> elements.

Additionally, I believe there should be a <tr> tag before the <td> elements (but still inside the forEach loop) as well as a corresponding </tr> closing tag after them (also still inside the forEach loop). That way, each time the loop runs to create a new item entry it will be creating the cells on a new row.

And lastly, it looks like according to the database structure for the fridge_products and products tables that we are querying from, the key names are actually 'prod_name', 'type', 'exp_dt', and 'qty', in order. Updating these should ensure that the correct object values are being read and printed into the table.

So to summarize, the final version should look something like this:

```
<% results.forEach(function(r) { %>
  <tr>
    <td><%= r.prod_name%></td>
    <td><%= r.type%></td>
    <td><%= r.exp_dt%></td>
    <td><%= r.qty%></td>
  </tr>
<% }); %>
```

Reply:
Thanks for the clarification, I have added this.